src/OrbitGl/PagefaultTrack.cpp

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "PagefaultTrack.h"
+
+#include "Geometry.h"
+#include "TextRenderer.h"
+#include "TimeGraph.h"
+#include "TimeGraphLayout.h"
+#include "Viewport.h"
+
+namespace orbit_gl {
+
+PagefaultTrack::PagefaultTrack(CaptureViewElement* parent, TimeGraph* time_graph,
+                               orbit_gl::Viewport* viewport, TimeGraphLayout* layout,
+                               std::array<std::string, kBasicPagefaultTrackDimension> series_names,
+                               const orbit_client_model::CaptureData* capture_data,
+                               uint32_t indentation_level)
+    : Track(parent, time_graph, viewport, layout, capture_data, indentation_level),
+      major_pagefault_track_{std::make_shared<BasicPagefaultTrack>(
+          this, time_graph, viewport, layout, "Major Pagefault Track", series_names, capture_data,
+          indentation_level + 1)},
```

Comment:
`indentation_level_` is the member of `Track`. This is used to determine the font size used in each track (e.g., track label tab, legend text, annotation text).

I prefer to keep initializing it in the constructor such that when creating a track. Are you suggesting set the indentation level in the draw method?

Reply:
Yes, I think that if we are only using it for drawing, should be there.

Reply:
I think the indentation level should be some property of the track, and it will determines how the track looks like. And it should not be changed anymore after we create the track and initialize it. If we set it every time when calling the draw method, then it doesn't need to be a class member.

Reply:
Ok, let's do it in that way :)